magma/clock.py

```diff
@@ -182,7 +182,8 @@ class EnableType(_BitType):
 
 
 def ClockInterface(has_enable=False, has_reset=False, has_set=False,
-                   has_ce=False, has_async_reset=False):
+                   has_ce=False, has_async_reset=False,
```

Comment:
I think it would be good to decouple `has_reset` from `reset_type` (those could be the exact parameters to this function). Seems like you wouldn't want to have multiple resets anyway (perhaps I'm missing some use case). Also, this would change the interface of this function, so current users may be affected.